lib/api/unittest/CDataFrameAnalyzerFeatureImportanceTest.cc

```diff
@@ -233,53 +221,54 @@ BOOST_FIXTURE_TEST_CASE(testRunBoostedTreeRegressionFeatureImportanceNoImportanc
     std::size_t topShapValues{4};
     auto results{runRegression(topShapValues, {10.0, 0.0, 0.0, 0.0}, 10.0)};
 
-    double c1, c2, c3, c4;
-    double prediction;
     for (const auto& result : results.GetArray()) {
         if (result.HasMember("row_results")) {
-            c1 = result["row_results"]["results"]["ml"]["shap.c1"].GetDouble();
-            c2 = result["row_results"]["results"]["ml"]["shap.c2"].GetDouble();
-            c3 = result["row_results"]["results"]["ml"]["shap.c3"].GetDouble();
-            c4 = result["row_results"]["results"]["ml"]["shap.c4"].GetDouble();
-            prediction = result["row_results"]["results"]["ml"]["c5_prediction"].GetDouble();
-            BOOST_REQUIRE_CLOSE(c1, prediction, 99); //c1 explain 99% of the prediction value
-            BOOST_REQUIRE_SMALL(c2, 1e-8);
-            BOOST_REQUIRE_SMALL(c3, 1e-8);
-            BOOST_REQUIRE_SMALL(c4, 1e-8);
+            double c1{result["row_results"]["results"]["ml"]["shap.c1"].GetDouble()};
+            double c2{result["row_results"]["results"]["ml"]["shap.c2"].GetDouble()};
+            double c3{result["row_results"]["results"]["ml"]["shap.c3"].GetDouble()};
+            double c4{result["row_results"]["results"]["ml"]["shap.c4"].GetDouble()};
+            double prediction{
+                result["row_results"]["results"]["ml"]["c5_prediction"].GetDouble()};
+            // c1 explain 97% of the prediction value, i.e. the difference from the prediction is less than 1%.
+            BOOST_REQUIRE_CLOSE(c1, prediction, 3.0);
+            BOOST_REQUIRE_SMALL(c2, 0.25);
+            BOOST_REQUIRE_SMALL(c3, 0.25);
+            BOOST_REQUIRE_SMALL(c4, 0.25);
```

Comment:
These numbers are slightly unstable. The reason they were 0 before before is that none of these features were selected at all for training because of the threshold on MIC. Once they get selected there is scope for them to be non-zero. Since these are still so much smaller than c1, 0.25 vs ~100, I figured it was ok to just allow some leeway.

Reply:
:+1: